### PR TITLE
Fix node ephemeral storage metrics

### PR DESCRIPTION
**Pull request by kawych**
**What this PR does**
Fixes ephemeral storage metrics, that used to be not exported or exported as 0.

/cc @olagacek

### CODE REVIEW

File: metrics/sources/summary/summary.go [hunk 1]
```diff
@@ -267,6 +267,8 @@ func (this *summaryMetricsSource) decodeEphemeralStorageStats(metrics *MetricSet
 		return
 	}
 	this.addIntMetric(metrics, &MetricEphemeralStorageUsage, storage.UsedBytes)
+	this.addIntMetric(metrics, &MetricNodeEphemeralStorageCapacity, storage.CapacityBytes)
```
Comment (x13n) [inline metrics/sources/summary/summary.go:270]: I don't see this metric in public docs: https://cloud.google.com/monitoring/api/metrics_kubernetes

File: metrics/processors/node_autoscaling_enricher.go [hunk 2]
```diff
@@ -73,9 +72,7 @@ func (this *NodeAutoscalingEnricher) Process(batch *core.DataBatch) (*core.DataB
 			setFloat(metricSet, &core.MetricNodeMemoryCapacity, float64(capacityMem.Value()))
 			setFloat(metricSet, &core.MetricNodeMemoryAllocatable, float64(allocatableMem.Value()))
 
-			if storageExist && allocatableStorageExist {
-				setFloat(metricSet, &core.MetricNodeEphemeralStorageCapacity, float64(capacityEphemeralStorage.Value()))
```
Comment (x13n) [inline metrics/processors/node_autoscaling_enricher.go:77]: Here you are removing this and later you're trying to process this metric. Why?

File: metrics/sources/summary/summary.go (hunk 1 above)
Comment (kawych) [reply metrics/sources/summary/summary.go:270]: It's there: node/ephemeral_storage/total_bytes

case core.MetricNodeEphemeralStorageCapacity.MetricDescriptor.Name:
	point := sink.intPoint(timestamp, timestamp, value.IntValue)
	return createTimeSeries("k8s_node", nodeLabels, ephemeralstorageTotalBytesMD, point)

File: metrics/processors/node_autoscaling_enricher.go (hunk 2 above)
Comment (kawych) [reply metrics/processors/node_autoscaling_enricher.go:77]: I've moved this metrics from based on API Server to based on Summary API. Can you clarify what's the concern?

File: metrics/sources/summary/summary.go (hunk 1 above)
Comment (x13n) [reply metrics/sources/summary/summary.go:270]: No, this is ephemeral_storage/node_capacity: https://github.com/Stackdriver/heapster/blob/master/metrics/core/metrics.go#L662

Comment (x13n) [reply metrics/sources/summary/summary.go:270]: node/ephemeral_storage/total_bytes should be a total usage, not the capacity.

File: metrics/processors/node_autoscaling_enricher.go (hunk 2 above)
Comment (x13n) [reply metrics/processors/node_autoscaling_enricher.go:77]: Capacity and allocatable values are not available in the Summary API as far as I know. Did you verify they are there for ephemeral storage?

File: metrics/sources/summary/summary.go (hunk 1 above)
Comment (kawych) [reply metrics/sources/summary/summary.go:270]: The metric you point to is just an implementation detail. It's one of "core" metrics, which are an artifact from the times when Heapster supported multiple sinks, and each of them could specify their own metric naming. This metric is mapped to "node/ephemeral_storage/total_bytes" in Stackdriver sink (in a code snippet I included in the comment above).

"node/ephemeral_storage/total_bytes should be a total usage, not the capacity." - I don't really understand this comment. We have a separate metric for usage, total_bytes is a node capacity. We also used a capacity before this PR, I'm not changing it here.

File: metrics/processors/node_autoscaling_enricher.go (hunk 2 above)
Comment (kawych) [reply metrics/processors/node_autoscaling_enricher.go:77]: Yes, they are available and the same as capactiy and allocatable that API Server uses.

**Review by x13n — APPROVED** [review]

**Comment by piosz** [comment]
Can this be merged?

**Comment by x13n** [comment]
I guess so. @kawych is out of office, so I'll just go ahead and merge this myself.

**Pull request merged**